docs/OCI_COMPLIANCE.md

```diff
@@ -0,0 +1,256 @@
+- Direct sysfs monitoring
+
+### 3. Configuration
+
+Container configuration via `config.json` supports:
+
+- Root filesystem specification ([spec](https://github.com/opencontainers/runtime-spec/blob/main/config.md#root))
+- Process arguments (firmware binary name) ([spec](https://github.com/opencontainers/runtime-spec/blob/main/config.md#process))
+- Annotations (remoteproc.name for processor selection) ([spec](https://github.com/opencontainers/runtime-spec/blob/main/config.md#annotations))
+- OCI version compatibility ([spec](https://github.com/opencontainers/runtime-spec/blob/main/config.md#specification-version))
+
+### 4. Namespace Isolation
+
+**Standard OCI**: Must support Linux namespaces for process isolation ([OCI Config Spec - Linux Namespaces](https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#namespaces)).
+
+**Remoteproc Runtime**: **No namespace isolation implemented**.
+
+**Rationale**: Firmware runs on a physically separate processor with its own address space and execution context. The proxy process that manages the processor lifecycle runs in the host's namespaces. Traditional Linux namespace isolation is meaningless for auxiliary processor firmware.
+
+**Impact**: Security boundaries are hardware-enforced (separate processors) rather than software-enforced (Linux namespaces).
+
+### 5. Resource Management and Cgroups
+
+**Standard OCI**: Requires cgroups support for resource limits ([OCI Config Spec - Linux Control Groups](https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#control-groups)).
+
+**Remoteproc Runtime**: **No cgroups support**.
+
+**Rationale**: The auxiliary processor has dedicated hardware resources independent of the Linux host. Resource limits on the host-side proxy process are irrelevant since the actual workload executes on separate silicon with its own memory, CPU cycles, and peripherals.
+
+**Impact**: Resource management must be handled at the hardware level or through processor-specific configuration mechanisms, not via Linux cgroups.
+
+### 6. Filesystem and Mounts
+
+**Standard OCI**: Comprehensive filesystem support ([OCI Config Spec - Mounts](https://github.com/opencontainers/runtime-spec/blob/main/config.md#mounts)).
+
+**Remoteproc Runtime**: **Limited filesystem usage**:
+
+- Rootfs is read to extract firmware binary during create phase
+- Firmware copied to `/lib/firmware/` with unique timestamped name
+- Proxy process does **not** chroot or mount the rootfs
+- No bind mounts, no mount propagation, no mount options
+
+**Rationale**: The firmware binary is loaded by the Linux kernel's remoteproc framework into the processor's memory. The rootfs serves as a packaging mechanism only.
+
+**Impact**:
+
+- Container images can be minimal (single-file firmware)
+- No need for complex filesystem layouts or mount management
+- Firmware persistence handled by copying to system firmware directory
```

Comment:
Not sure about this section, it feels like it's trying to spin a constraint as a positive
```suggestion
- The firmware file in the rootfs is the only source of filesystem state passed from the container engine to the remote processor.
- Container engine provides no mechanism for filesystem sharing between host and remote processor
- Firmware persistence handled by copying to system firmware directory
```